src/Avalonia.Controls.DataGrid/DataGridRowGroupHeader.cs

```diff
@@ -472,5 +473,11 @@ internal void UpdateTitleElements()
             }
         }
 
+        protected override void OnDetachedFromLogicalTree(LogicalTreeAttachmentEventArgs e)
+        {
+            base.OnDetachedFromLogicalTree(e);
+            OwningGrid.RemoveReferenceFromCollectionViewGroup(RowGroupInfo.CollectionViewGroup);
+            OwningGrid = null;
```

Comment:
This assumes that the grid will never get attached back to the logical tree.
If it does and the groups are kept, they just won't work correctly with this change.

(I'm writing this as a reminder to test this specific case, it's entirely possible that the groups are completely recreated... There are no unit tests validating that behavior.)

Reply:
Just tested it (with some additional output on DataGrid's `OnDetachedFromLogicalTree`).

Only the 'recollapsing' doesn't work, since the DataGrid is completely recreated.